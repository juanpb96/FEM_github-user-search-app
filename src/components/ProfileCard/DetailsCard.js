export const DetailsCard = ({ repos, followers, following }) => {
    return (
        <div className='[ details ] [ bg-primary br-10 mb-24 ]'>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Repos</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ repos }</p>
            </div>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Followers</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ followers }</p>
            </div>
            <div>
                <p className='[ mb-8 fs-xsmall ]'>Following</p>
                <p className='[ title-color fs-medium fw-700 ]'>{ following }</p>
            </div>
        </div>
    );
};
