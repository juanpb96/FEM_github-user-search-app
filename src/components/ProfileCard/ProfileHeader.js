export const ProfileHeader = ({ atName, created_at, name }) => {

    const joinDate = new Date(created_at);
    const [ day, month, year ] = [
        joinDate.getDate(),
        joinDate.toLocaleString('en-US', { month: 'short' }),
        joinDate.getFullYear()
    ];

    return (
        <header className='[ profile__header ]'>
            <h2 
                className='[ title-color fs-medium fw-700 ]'
                datatype='username'
            >{ name ? name : atName }</h2>
            <p 
                className='[ text-color-primary fw-400 fs-small ]'
                datatype='join-date'
            >{ `Joined at ${ day } ${ month } ${ year }` }</p>
            <p 
                className='[ text-color-blue fw-400 ]'
                datatype='at'
            >@{ atName }</p>
        </header>
    );
}
