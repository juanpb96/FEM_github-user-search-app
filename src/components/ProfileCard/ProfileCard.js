import { useSelector } from "react-redux";
import { DetailsCard } from "./DetailsCard";
import { ListIconItem } from "./ListIconItem";
import { ProfileHeader } from "./ProfileHeader";

export const ProfileCard = () => {
    const { 
        login: atName,
        avatar_url: avatar,
        bio,
        blog,
        company,
        followers,
        following,
        created_at,
        location,
        name,
        public_repos: repos,
        twitter_username: twitter
    } = useSelector( state => state.user );

    const icons = [
        { name: 'location', value: location },
        { name: 'webpage', value: blog },
        { name: 'twitter', value: twitter },
        { name: 'company', value: company }
    ];

    return (
        <article className='[ profile ] [ bg-secondary card-border ]'>
            
            <ProfileHeader 
                atName={ atName }
                avatar={ avatar }
                created_at={ created_at }
                name= { name }
            />

            <p 
                className='[ profile__description ][ mb-24 text-color-secondary fs-small fw-400 lh-large ]'
            >{ bio ? bio : 'This profile has no bio' }</p>
            
            <DetailsCard 
                repos={ repos }
                followers={ followers }
                following={ following }
            />

            <ul className='[ list ] [ grid-flow ] [ no-bullets ]'>
                {
                    icons.map( item => (
                        <ListIconItem
                            key={ item.name } 
                            name={ item.name }
                            value={ item.value }
                        />
                    ))
                }
            </ul>
        </article>
    );
};
