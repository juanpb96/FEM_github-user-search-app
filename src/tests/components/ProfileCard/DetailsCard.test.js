import { shallow } from "enzyme";
import { DetailsCard } from "../../../components/ProfileCard/DetailsCard";

describe('Test <DetailsCard />', () => { 
    test('should render correctly', () => { 
            const wrapper = shallow(
                <DetailsCard 
                    repos={10}
                    followers={20}
                    following={350}
                />
            );

            expect(wrapper).toMatchSnapshot();
            
            const [ 
                repos,
                followers,
                following
            ] = wrapper.find('.title-color').map(e => +e.text());

            expect(repos).toBe(10);
            expect(followers).toBe(20);
            expect(following).toBe(350);
    });   
});