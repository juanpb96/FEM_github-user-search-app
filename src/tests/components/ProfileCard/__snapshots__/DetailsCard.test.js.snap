// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <DetailsCard /> should render correctly 1`] = `
<section
  className="[ details ] [ bg-primary br-10 mb-24 ]"
>
  <div>
    <p
      className="[ mb-8 fs-xsmall ]"
    >
      Repos
    </p>
    <p
      className="[ title-color fs-medium fw-700 ]"
    >
      10
    </p>
  </div>
  <div>
    <p
      className="[ mb-8 fs-xsmall ]"
    >
      Followers
    </p>
    <p
      className="[ title-color fs-medium fw-700 ]"
    >
      20
    </p>
  </div>
  <div>
    <p
      className="[ mb-8 fs-xsmall ]"
    >
      Following
    </p>
    <p
      className="[ title-color fs-medium fw-700 ]"
    >
      350
    </p>
  </div>
</section>
`;
